bootstrap.sh

```diff
@@ -0,0 +1,175 @@
+#! /bin/bash
+banner() {
+	echo "|------------------------------------------|"
+	echo "|----- Welcome to the redox bootstrap -----|"
+	echo "|------------------------------------------|"
+}
+
+osx()
+{
+	echo "Detected OSX!"
+	temp=$(brew --version)
+	if [ -n "$temp" ]; then
+		echo "Homebrew detected! Now updating..."
+		brew update
+		echo "Now installing git..."
+		brew install git
+		echo "Now installing virtualbox..."
+		brew cask install virtualbox
+	else
+		echo "Homebrew does not appear to be installed! Would you like me to install it?"
+		printf "(Y/n): "
+		#For now assume yes
+		ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+	fi
+	echo "Cloning Redox repo"
+	git clone -b $1 --recursive https://github.com/redox-os/redox.git
+	sh redox/setup/osx-homebrew.sh
+	sh redox/setup/binary.sh
+}
+
+archLinux()
+{
+	echo "Detected Arch Linux, mah fav"
+	echo "Updating system..."
+	sudo pacman -Syu
+	echo "Installing git..."
+	sudo pacman -S git
+	echo "Cloning Redox repo"
+    if [ "$2" == "qemu" ]; then
+        echo "Installing QEMU..."
+        sudo pacman -S qemu
+    fi
+	git clone -b $1 --recursive https://github.com/redox-os/redox.git
+	sh redox/setup/arch.sh
+	sh redox/setup/binary.sh
+}
+
+ubuntu()
+{
+	echo "Detected Ubuntu/Debian"
+	echo "Updating system..."
+	sudo apt-get upgrade
+	echo "Installing git..."
+	sudo apt-get -y install git
+    if [ "$2" == "qemu" ]; then
+        echo "Installing QEMU..."
+        sudo apt-get -y install qemu-system-x86 qemu-kvm
+    fi
+	echo "Cloning Redox repo"
+	git clone -b $1 --recursive https://github.com/redox-os/redox.git
+    sudo apt-get -y install build-essential libc6-dev-i386 nasm curl file
```

Comment:
@NobbZ 
Okay so a command line option for this would not be difficult to implement at all then!
Also to address the -y, I attempted to run the script in a hastily prepared chroot and running the script it did not wait for me to press y it just assumed n and exited so that is why they are there. I guess I messed something up with that, I will drop them.